source/common/http/async_client_impl.cc

```diff
@@ -264,6 +282,22 @@ void AsyncRequestImpl::onTrailers(HeaderMapPtr&& trailers) {
 }
 
 void AsyncRequestImpl::onReset() {
+  if (!cancelled_) {
+    // Add tags about reset.
+    child_span_->setTag(Tracing::Tags::get().Error, Tracing::Tags::get().True);
+    child_span_->setTag(Tracing::Tags::get().ErrorReason, "Reset");
+  }
+
+  // Finalize the span based on whether we received a response or not
```

Comment:
Can this be consolidated? It seems like this can be a single call where we either have headers or not, trailers or not, etc.